tools/accuracy_checker/accuracy_checker/metrics/coco_orig_metrics.py

```diff
@@ -202,13 +161,82 @@ def _convert_data_to_coco_format(
         return coco_data_to_store
 
     @staticmethod
-    def _reload_results_to_coco_class(coco, coco_data_to_store):
+    def _calculate_area(data):
+        width = data.x_maxs - data.x_mins + 1
+        height = data.y_maxs - data.y_mins + 1
+        area = width * height
+
+        return area
+
+    def _prepare_data_for_annotation_file(
+            self, annotations, dataset_label_map, iou_type='bbox'
+    ):
+        coco_annotation_to_store = []
+        coco_category_to_store = []
+        coco_image_to_store = []
+        label_map = set()
+        count = 0
+        for annotation in annotations:
+            annotation_data_to_store = []
+            cur_name = annotation.identifier
+            cur_name = os.path.basename(cur_name)
+            cur_img_id = int(cur_name.split(".")[0])
+
+            labels = annotation.labels.tolist()
+            areas = self._calculate_area(annotation).tolist()
+            iscrowds = annotation.metadata.get('iscrowd')
+
+            for cur_cat, iscrowd, area in zip(
+                    labels, iscrowds, areas
+            ):
+                annotation_data_to_store.append({
+                    'id': count,
+                    'image_id': cur_img_id,
+                    'category_id': cur_cat,
+                    '_image_name_from_dataset': cur_name,
+                    'iscrowd': iscrowd,
+                    'area': area
+                })
+                count += 1
+                label_map.add(cur_cat)
+
+            iou_specific_converter = iou_specific_processing.get(iou_type)
+            if iou_specific_converter is None:
```

Comment:
please make sure that new conversion code works correctly also for `segm` and `keypoints` iou_types

Reply:
Processing of segm and keypoints types have been added. Please review again